px4_ros2_cpp/src/navigation/experimental/local_navigation_interface.cpp

```diff
@@ -0,0 +1,201 @@
+/****************************************************************************
+ * Copyright (c) 2023 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+
+#include <rclcpp/rclcpp.hpp>
+#include <px4_ros2/navigation/experimental/local_navigation_interface.hpp>
+
+namespace px4_ros2
+{
+
+LocalNavigationInterface::LocalNavigationInterface(
+  rclcpp::Node & node, const uint8_t pose_frame,
+  const uint8_t velocity_frame)
+: _node(node),
+  _pose_frame(pose_frame),
+  _velocity_frame(velocity_frame)
+{
+  // Check that selected pose and velocity reference frames are valid
+  auto it_pose_frame = std::find(
+    std::begin(_available_pose_frames), std::end(
+      _available_pose_frames), pose_frame);
+  if (it_pose_frame == std::end(_available_pose_frames)) {
+    RCLCPP_ERROR(
+      node.get_logger(), "Failed to initialize LocalNavigationInterface: invalid pose reference frame %i.",
+      pose_frame);
+    rclcpp::shutdown();
+  }
+
+  auto it_vel_frame =
+    std::find(
+    std::begin(_available_velocity_frames), std::end(
+      _available_velocity_frames), velocity_frame);
+  if (it_vel_frame == std::end(_available_velocity_frames)) {
+    RCLCPP_ERROR(
+      node.get_logger(), "Failed to initialize LocalNavigationInterface: invalid velocity reference frame %i.",
+      velocity_frame);
+    rclcpp::shutdown();
+  }
+
+  _aux_local_position_pub = node.create_publisher<AuxLocalPosition>(AUX_LOCAL_POSITION_TOPIC, 10);
+
+}
+
+int LocalNavigationInterface::update(const LocalPositionEstimate & local_position_estimate) const
+{
+  // Run basic sanity checks on local position estimate
+  if (_checkEstimateEmpty(local_position_estimate)) {
+    RCLCPP_WARN(_node.get_logger(), "Estimate values are all empty.");
```

Comment:
I wonder whether `WARN` is the right level for this. While it is bad, if a user sends an invalid struct, there's a good chance they're continually sending it at a high rate, which would spam the logs. That might be a good thing too, but we could also make this log `DEBUG` level (or even remove it) and rely more on the return value for error catching. What do you think?

Reply:
I see your point. I'll set it to DEBUG in the meantime.